src/wallet/rpcwallet.cpp

```diff
@@ -597,6 +602,92 @@ UniValue listaddresses(const UniValue& params, bool fHelp)
+                mnemonic_taddrs.push_back(keyIO.EncodeDestination(keyId));
+                hasData = true;
+            }
+        }
+        if (hasData) {
+            entry.pushKV("transparent", mnemonic_taddrs);
+        }
+
+        // sapling
+        hasData |= add_sapling(saplingAddresses, PaymentAddressSource::MnemonicHDSeed, entry);
+
+        // unified
+        // here, we want to use the information in mapUfvkAddressMetadata to report all the unified addresses
+        UniValue unified_groups(UniValue::VARR);
+        auto hdChain = pwalletMain->GetMnemonicHDChain();
+        for (const auto& [ufvkid, addrmeta] : pwalletMain->mapUfvkAddressMetadata) {
+            auto account = pwalletMain->GetUnifiedAccountId(ufvkid);
+            if (account.has_value() && hdChain.has_value()) {
+                // double-check that the ufvkid we get from address metadata is actually
+                // associated with the mnemonic HD chain
+                auto ufvkCheck = pwalletMain->mapUnifiedAccountKeys.find(
+                    std::make_pair(hdChain.value().GetSeedFingerprint(), account.value())
+                );
+                if (ufvkCheck != pwalletMain->mapUnifiedAccountKeys.end() && ufvkCheck->second == ufvkid) {
+                    auto hdKeyPath = Zip32AccountKeyPath(Params().BIP44CoinType(), account.value());
+                    UniValue unified_group(UniValue::VOBJ);
+                    unified_group.pushKV("zip32KeyPath", hdKeyPath);
+
+                    UniValue unified_addrs(UniValue::VARR);
+                    auto ufvk = pwalletMain->GetUnifiedFullViewingKey(ufvkid).value();
+                    for (const auto& [j, receiverTypes] : addrmeta.GetKnownReceiverSetsByDiversifierIndex()) {
+                        // We know we can use std::get here safely because we previously
+                        // generated a valid address for this diversifier & set of
+                        // receiver types.
+                        UniValue addrEntry(UniValue::VOBJ);
+                        auto addr = std::get<std::pair<libzcash::UnifiedAddress, diversifier_index_t>>(
+                            ufvk.Address(j, receiverTypes)
+                        );
+                        UniValue receiverTypesEntry(UniValue::VARR);
+                        for (auto t : receiverTypes) {
+                            switch(t) {
+                                case ReceiverType::P2PKH:
+                                    receiverTypesEntry.push_back("P2PKH");
+                                    break;
+                                case ReceiverType::P2SH:
+                                    receiverTypesEntry.push_back("P2SH");
+                                    break;
+                                case ReceiverType::Sapling:
+                                    receiverTypesEntry.push_back("Sapling");
+                                    break;
```

Comment:
Why does this not include Orchard?

Reply:
Orchard support will be added later; its receiver type isn't yet defined: https://github.com/zcash/zcash/blob/d724c812890e7d27f581c6959a7f1e8af3a0b79b/src/zcash/address/unified.h#L26

Reply:
This PR is based on master, not `feature/wallet_orchard`. As such, UAs don't have Orchard yet. This is an ordering bug that we can fix during the RC.

Reply:
No problem, but I don't see an issue in the DAG to add it in the RC period.

Reply:
#5710 will merge today, and since adding Orchard support is fairly trivial hopefully this can be rebased on `master` after that happens and everything can get in for the RC on Monday.

Reply:
#5710 is merged, so yeah, add Orchard support when you rebase it.